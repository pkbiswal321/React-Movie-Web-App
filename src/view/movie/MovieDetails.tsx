import React from 'react'
import './style/MovieDetails.css'

export interface movieDataDetails {
  id: number
  name: string
  show: {
    image?: {
      medium: string
    }
  }
  url: string
}

export interface MovieDetailsProps {
  movieData: Array<movieDataDetails>
}

const MovieDetails = ({ movieData }: MovieDetailsProps) => {
  // console.log('movieData', movieData)
  return (
    <div className="container">
      {movieData.length > 0 &&
        movieData.map((item) => {
          return (
            <div key={item.id}>
              <div className="card">
                <div className="circle">
                  <img src={item.show.image?.medium} alt="" />
                </div>
                <div className="content">
                  <h5>{item.name}</h5>
                  <a href={item.url} target="_blank">
                    Buy Now
                  </a>
                </div>
              </div>
            </div>
          )
        })}
    </div>
  )
}

export default MovieDetails
