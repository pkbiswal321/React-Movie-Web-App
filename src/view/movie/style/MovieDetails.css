* {
  margin: 0px;
  padding: 0px;
  box-sizing: border-box;
}

body {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  background: linear-gradient(45deg, rgb(122, 231, 226), #bb02ff);
  background-attachment: fixed;
}

.container {
  position: relative;
  max-width: 1150px;
  padding: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
}

.container .card {
  position: relative;
  width: 300px;
  height: 420px;
  background: #ffff;
  margin: 20px;
  border-radius: 20px;
  overflow: hidden;
  box-shadow: 0 15px 25px rgb(0, 0, 0, 0.2);
  transition: 0.5s;
}

.container:hover .card {
  filter: blur(1px);
  transform: scale(0.9);
  opacity: 0.9;
}

.container .card:hover {
  filter: blur(0px);
  transform: scale(1.1);
  opacity: 1;
}

.container .card .circle {
  position: absolute;
  width: 100%;
  height: 100%;
  background: black;
  clip-path: circle(250px at center 0);
  text-align: center;
}

.container .card .circle img {
  width: 100%;
}

.container .card .content {
  position: absolute;
  bottom: 30px;
  padding: 20px;
  text-align: center;
  width: 100%;
}

.container .card .content h5 {
  color: #666;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  /* text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap; */
}

.container .card .content a {
  position: relative;
  display: inline-block;
  padding: 10px 20px;
  background: black;
  color: #fff;
  border-radius: 40px;
  text-decoration: none;
  margin-top: 20px;
}
