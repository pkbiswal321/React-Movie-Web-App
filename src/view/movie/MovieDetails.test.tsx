import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import MovieDetails, { MovieDetailsProps, movieDataDetails } from "./MovieDetails";

export const mockdata: Array<movieDataDetails> = [
    {
        id: 1979600,
        name: "Minute Whambam 90's Ride",
        show: {
            image: {
                medium: "https://static.tvmaze.com/uploads/images/medium_portrait/105/264401.jpg"
            }
        },
        url: "https://www.tvmaze.com/episodes/1979386/patis-mexican-table-9x09-cooking-for-my-crew-in-sonora",
    },
    {
        id: 1979601,
        name: "Minute Whambam 90's Ride",
        show: {
            image: {
                medium: "https://static.tvmaze.com/uploads/images/medium_portrait/105/264401.jpg"
            }
        },
        url: "https://www.tvmaze.com/episodes/1979386/patis-mexican-table-9x09-cooking-for-my-crew-in-sonora",
    },
]

describe("MovieDetails", () => {
    let wrapper: any
    beforeAll(() => {
        wrapper = shallow(<MovieDetails movieData={mockdata} />);
    });    

    it("matches snapshot", () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    // it('render component div', () => {
    //     expect(wrapper.firstChild.classList.contains('div.container')).toBe(true)
    // });

});