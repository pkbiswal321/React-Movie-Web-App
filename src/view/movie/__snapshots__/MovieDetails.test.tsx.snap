// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails matches snapshot 1`] = `
<div
  className="container"
>
  <div
    key="1979600"
  >
    <div
      className="card"
    >
      <div
        className="circle"
      >
        <img
          alt=""
          src="https://static.tvmaze.com/uploads/images/medium_portrait/105/264401.jpg"
        />
      </div>
      <div
        className="content"
      >
        <h5>
          Minute Whambam 90's Ride
        </h5>
        <a
          href="https://www.tvmaze.com/episodes/1979386/patis-mexican-table-9x09-cooking-for-my-crew-in-sonora"
          target="_blank"
        >
          Buy Now
        </a>
      </div>
    </div>
  </div>
  <div
    key="1979601"
  >
    <div
      className="card"
    >
      <div
        className="circle"
      >
        <img
          alt=""
          src="https://static.tvmaze.com/uploads/images/medium_portrait/105/264401.jpg"
        />
      </div>
      <div
        className="content"
      >
        <h5>
          Minute Whambam 90's Ride
        </h5>
        <a
          href="https://www.tvmaze.com/episodes/1979386/patis-mexican-table-9x09-cooking-for-my-crew-in-sonora"
          target="_blank"
        >
          Buy Now
        </a>
      </div>
    </div>
  </div>
</div>
`;
